src/assets/CollectiblesController.test.ts

```diff
@@ -1373,4 +1373,202 @@ describe('CollectiblesController', () => {
       });
     });
   });
+
+  describe('findCollectibleByAddressAndTokenId', () => {
+    const mockTransactionId = '60d36710-b150-11ec-8a49-c377fbd05e27';
+
+    const mockCollectible = {
+      address: '0x02',
+      tokenId: '1',
+      name: 'name',
+      image: 'image',
+      description: 'description',
+      standard: 'standard',
+      favorite: false,
+    };
+
+    const expectedMockCollectible = {
+      address: '0x02',
+      description: 'description',
+      favorite: false,
+      image: 'image',
+      name: 'name',
+      standard: 'standard',
+      tokenId: '1',
+      transactionId: mockTransactionId,
+    };
+
+    it('should return undefined when the collectible does not exist', async () => {
+      expect(
+        collectiblesController.findCollectibleByAddressAndTokenId(
+          mockCollectible.address,
+          mockCollectible.tokenId,
+        )?.collectible,
+      ).toBeUndefined();
+    });
+
+    it('should return the collectible by the address and tokenId with the prop transactionId', async () => {
+      await collectiblesController.addCollectible(
+        mockCollectible.address,
+        mockCollectible.tokenId,
+        mockCollectible,
+      );
+
+      collectiblesController.updateCollectibleByAddressAndTokenId({
+        ...mockCollectible,
+        transactionId: mockTransactionId,
+      });
+
+      expect(
+        collectiblesController.findCollectibleByAddressAndTokenId(
+          mockCollectible.address,
+          mockCollectible.tokenId,
+        )?.collectible,
+      ).toStrictEqual(expectedMockCollectible);
+    });
+  });
+
+  describe('updateCollectibleByAddressAndTokenId', () => {
+    const mockTransactionId = '60d36710-b150-11ec-8a49-c377fbd05e27';
+
+    const mockCollectible = {
+      address: '0x02',
+      tokenId: '1',
+      name: 'name',
+      image: 'image',
+      description: 'description',
+      standard: 'standard',
+      favorite: false,
+    };
+
+    const expectedMockCollectible = {
+      address: '0x02',
+      description: 'description',
+      favorite: false,
+      image: 'image',
+      name: 'name',
+      standard: 'standard',
+      tokenId: '1',
+      transactionId: mockTransactionId,
+    };
+    it('should update the collectible when the collectible exist', async () => {
+      const { selectedAddress, chainId } = collectiblesController.config;
+
+      await collectiblesController.addCollectible(
```

Comment:
Sort of the same argument as above: you shouldn't need to — and in fact it creates more of an isolated test if you don't — use `addCollectible` to add something to the state.